pipeline/test/test_pipeline.py

```diff
@@ -174,3 +211,75 @@ def test_scatter_gather(self):
     #
     #     self.assertEqual(self.read(output_file), msg)
     #     self.rm(output_file)
+
+
+class BatchTests(unittest.TestCase):
+    def pipeline(self):
+        url = os.environ.get('BATCH_URL')
+        if url is None:
+            raise Exception("BATCH_URL not defined in the environment.")
+        return Pipeline(backend=BatchBackend(url),
```

Comment:
I think `BatchBackend(os.environ['BATCH_URL'])` is fine, this is test code and a KeyError that references this line is pretty clear about what happened.